pkg/controller/v1alpha1/routes/route_controller.go

```diff
@@ -0,0 +1,436 @@
+		},
+	}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: selfSigned, Namespace: routeTrait.Namespace}, &issuer)
+	if err == nil {
+		return selfSigned, nil
+	}
+	issuer.Spec.SelfSigned = &certmanager.SelfSignedIssuer{}
+	if apierrors.IsNotFound(err) {
+		return selfSigned, r.Client.Create(ctx, &issuer)
+	}
+	return selfSigned, fmt.Errorf("get %s err %v", selfSigned, err)
+}
+
+func constructNginxIngress(routeTrait *standardv1alpha1.Route, issuer standardv1alpha1.TLS, service *runtimev1alpha1.TypedReference, port int32) *v1beta1.Ingress {
+
+	var annotations = make(map[string]string)
+
+	// SSL
+	var issuerAnn = "cert-manager.io/issuer"
+	if issuer.Type == standardv1alpha1.ClusterIssuer {
+		issuerAnn = "cert-manager.io/cluster-issuer"
+	}
+	annotations[issuerAnn] = issuer.IssuerName
+
+	// Rewrite
+	if routeTrait.Spec.RewriteTarget != "" {
+		annotations["ingress.kubernetes.io/rewrite-target"] = routeTrait.Spec.RewriteTarget
+	}
+
+	// Custom headers
+	var headerSnippet string
+	for k, v := range routeTrait.Spec.CustomHeaders {
+		headerSnippet += fmt.Sprintf("more_set_headers \"%s: %s\";\n", k, v)
+	}
+	if headerSnippet != "" {
+		annotations["nginx.ingress.kubernetes.io/configuration-snippet"] = headerSnippet
+	}
+	backend := routeTrait.Spec.Backend
+	if backend != nil {
+		// Backend protocol
+		if backend.Protocol != "" {
+			annotations["nginx.ingress.kubernetes.io/backend-protocol"] = backend.Protocol
+		} else {
+			annotations["nginx.ingress.kubernetes.io/backend-protocol"] = "HTTPS"
+		}
+
+		//Send timeout
+		if backend.SendTimeout != 0 {
+			annotations["nginx.ingress.kubernetes.io/proxy-send-timeout"] = strconv.Itoa(backend.SendTimeout)
+		}
+
+		//Read timeout
+		if backend.ReadTimeout != 0 {
+			annotations["nginx.ingress.kubernetes.io/proxy‑read‑timeout"] = strconv.Itoa(backend.ReadTimeout)
+		}
+	}
+
+	ingress := &v1beta1.Ingress{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       reflect.TypeOf(v1beta1.Ingress{}).Name(),
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        routeTrait.Name,
+			Namespace:   routeTrait.Namespace,
+			Annotations: annotations,
+			Labels:      oamServiceLabel,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         routeTrait.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+					Kind:               routeTrait.GetObjectKind().GroupVersionKind().Kind,
+					UID:                routeTrait.GetUID(),
+					Name:               routeTrait.GetName(),
+					Controller:         pointer.BoolPtr(true),
+					BlockOwnerDeletion: pointer.BoolPtr(true),
+				},
+			},
+		},
+	}
+	var ingressClass = "nginx"
+	ingress.Spec.IngressClassName = &ingressClass
+	ingress.Spec.TLS = []v1beta1.IngressTLS{
+		{
+			Hosts:      []string{routeTrait.Spec.Host},
+			SecretName: routeTrait.Name + "-cert",
+		},
+	}
+	if routeTrait.Spec.DefaultBackend != nil {
+		ingress.Spec.Backend = routeTrait.Spec.DefaultBackend
```

Comment:
Do we need both spec.Backend and Rules.IngressRueValue.Paths.backend in the rout.spec? Can we make them the same?

Reply:
No we can't, they are different concepts.

* `spec.Backend` is default backend, which will be used only the normal backend won't work.